railgun/test/integration/suite_test.go

```diff
@@ -73,9 +80,63 @@ func deployControllers(ctx context.Context, ready chan ktfkind.ProxyReadinessEve
 		u := event.URL
 		proxyReady <- u
 
-		cmd := exec.CommandContext(ctx, "go", "run", "../../main.go", "--kong-url", fmt.Sprintf("http://%s:8001", u.Hostname()))
+		// create a tempfile to hold the cluster kubeconfig that will be used for the controller
+		kubeconfig, err := ioutil.TempFile(os.TempDir(), "kubeconfig-")
+		if err != nil {
+			panic(err)
+		}
+		defer kubeconfig.Close()
+
+		// dump the kubeconfig from kind into the tempfile
+		generateKubeconfig := exec.CommandContext(ctx, "kind", "get", "kubeconfig", "--name", cluster.Name())
+		generateKubeconfig.Stdout = kubeconfig
+		generateKubeconfig.Stderr = os.Stderr
+		if err := generateKubeconfig.Run(); err != nil {
+			panic(err)
+		}
+
+		// set the default command which runs the current controller manager code
+		cmd := exec.CommandContext(ctx, "go", "run", "../../main.go",
```

Comment:
Given the context of how this is run currently in CI and the fact that it's labeled as a hack with a `FIXME` and we have a follow-up item to take care of this before GA:

https://github.com/Kong/kubernetes-ingress-controller/issues/1128

I'm really more inclined to leave this alone for now since I expect it will go away quickly, unless you feel strongly.

Reply:
Never mind. I forgot that `go test` sets the CWD for you.

Reply:
:+1: indeed, sets it to `test/integration` on each run.